assopy/models.py

```diff
@@ -135,39 +144,68 @@ def get_url_path(self):
 
 
 class UserManager(models.Manager):
+    def _create_user(
+            self, email, password, username=None, first_name='', last_name='',
+            token=False, active=False, assopy_id=None, is_admin=False):
+        if not username:
+            username = janrain.suggest_username_from_email(email)
+
+        if is_admin:
+            duser = auth.models.User.objects.create_superuser(username, email, password=password)
+        else:
+            duser = auth.models.User.objects.create_user(username, email, password=password)
 
-    def create_user(self, email, first_name='', last_name='', password=None, token=False, active=False, assopy_id=None, send_mail=True):
-        uname = janrain.suggest_username_from_email(email)
-        duser = auth.models.User.objects.create_user(uname, email, password=password)
         duser.first_name = first_name
         duser.last_name = last_name
         duser.is_active = active
         duser.save()
         user = User(user=duser)
+
         if token:
             user.token = str(uuid4())
         if assopy_id is not None:
             user.assopy_id = assopy_id
         user.save()
+
```

Comment:
I'm confused by all the `user`s here, but let's leave it as–is and then later maybe refactor everything to AssopyUser so that those names don't confuse people in the future. ;)

Reply:
Agree that it's confusing - and fully support the improve-things-one-step-at-a-time approach :)